crates/events/src/enclave_event/compute_request/trbfv.rs

```diff
@@ -0,0 +1,149 @@
+    pub struct Request {
+        /// BFV Params
+        params: Bytes,
+        /// Max number of ciphertexts
+        max_num_ciphertexts: u64,
+    }
+
+    #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+    pub struct Response {
+        /// The smudging noise shares
+        esi_sss: Vec<SensitiveBytes>,
+    }
+}
+
+pub mod gen_pk_share_and_sk_sss {
+    /// This module defines event payloads that will generate the public key share as well as the sk shamir secret shares to be distributed to other members of the committee.
+    /// This has been separated from the esi setup in order to be able to take advantage of parallelism
+    use crate::bytes::Bytes;
+    use e3_crypto::SensitiveBytes;
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+    pub struct Request {
+        /// BFV Params
+        params: Bytes,
+    }
+
+    #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+    pub struct Response {
+        /// PublicKey share for this node
+        pk_share: Bytes,
+        /// SecretKey Shamir Shares for other parties
+        sk_sss: Vec<SensitiveBytes>,
+    }
+}
+
+pub mod gen_decryption_key {
+    /// This module defines event payloads that will generate the decryption key material to create a decryption share
+    use e3_crypto::SensitiveBytes;
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+    pub struct Request {
+        /// All collected secret key shamir shares
+        sk_sss_collected: Vec<SensitiveBytes>,
+        /// All collected smudging noise shamir shares
+        esi_sss_collected: Vec<SensitiveBytes>,
+    }
+
+    #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+    pub struct Response {
+        /// A single summed polynomial for this nodes secret key.
+        sk_poly_sum: SensitiveBytes,
+        /// A single summed polynomial for this partys smudging noise
+        es_poly_sum: SensitiveBytes,
+    }
+}
+
+pub mod gen_decryption_share {
+    /// This module defines event payloads that will generate a decryption share for the given ciphertext for this node
+    use crate::bytes::Bytes;
+    use e3_crypto::SensitiveBytes;
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+    pub struct Request {
+        /// BFV Params
+        params: Bytes,
+        /// Ciphertext to decrypt
+        ciphertext: Bytes,
+        /// A single summed polynomial for this nodes secret key.
+        sk_poly_sum: SensitiveBytes,
+        /// A single summed polynomial for this partys smudging noise
+        es_poly_sum: SensitiveBytes,
+    }
+
+    #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+    pub struct Response {
+        /// The decryption share for the given ciphertext
+        d_share_poly: Bytes,
+    }
+}
+
```

Comment:
are we missing `d_share_polys` aggregation before threshold decrypting?

Reply:
That occurs in another actor as it is not a cryptographic operation just aggregation